src/extensions/default/bramble/lib/xhr/XHRShim.js

```diff
@@ -32,77 +32,81 @@
 
     function XMLHttpRequestLiveDev() {
         var xhr = new XMLHttpRequest();
-        var requestUrl;
-        var abortCalled = false;
+        // Internally store properties normally attached to XHR
+        var _requestUrl;
+        var _abortCalled = false;
+        var _readyState = 0;
+        var _status = 0;
+        var _statusText = "";
+        var _response = "";
+        var _responseText = "";
+        var _responseType;
+        var _onreadystatechange;
+        var _onerror;
+        var _onload;
+        var self = this;
 
         var $open = xhr.open;
         // The async parameter for XHRs to the local file system
         // is ignored as we cannot support synchronous requests
         // and async is implicit
-        xhr.open = function(method, url, async, user, password) {
+        this.open = function(method, url, async, user, password) {
```

Comment:
I'd change all uses of `this` to `self` so you are consistent throughout this file.